### DISCUSSION TITLE
Dont working bot.load_extension

### DISCUSSION THREAD

**Discussion by neJok** [Rapptz/discord.py#7740] [Q&A] [answered]
My code:
![изображение_2022-03-21_133207](https://user-images.githubusercontent.com/84874782/159244116-66b0a592-7dce-42bc-90aa-d8bc9a8c822a.png)
Error
![изображение_2022-03-21_133241](https://user-images.githubusercontent.com/84874782/159244205-4c6ef746-6fb5-4f79-a841-5f0e23fdb485.png)
How fix it? Its not async

**Comment by AbstractUmbra** [comment, marked as answer]
There have been [extensive changes](https://github.com/Rapptz/discord.py/commit/a1c618215e7c3222e07093b50f166bcaf53dcfc8) on the current `master` branch of the repository, namely making most of the extension methods of the bot asynchronous.

This means that such things as:
`Bot.add_cog()`
`Bot.remove_cog()`
`Bot.load_extension`
`Bot.unload_extension`
and `Bot.reload_extension()` are now coroutine methods.

You can see more in a [gist that Danny wrote up on it](https://gist.github.com/Rapptz/6706e1c8f23ac27c98cee4dd985c8120).

**Reply by neJok** [reply]
How i can do async Bot.reload_extension() in code? await Bot.reload_extension() dont working.

**Reply by AbstractUmbra** [reply]
[We provided a `setup_hook` method](https://github.com/Rapptz/discord.py/commit/68b7ee6b99ff942c0a7138017680f0566f4fd1d7) within the `Bot` class so that our users have a valid async startup context to use it in.

The simplest method to using it, is to subclass `commands.Bot`:
```py
class MyBot(commands.Bot):
  async def setup_hook(self) -> None:
    await bot.load_extension("some_extension")
```
and use this class.

As for reloading extensions, they can be done within commands as per usual.

If this is unrelated to your issue, consider joining the [discord server](https://discord.gg/dpy) and asking in our help channels.

**Reply by neJok** [reply]
new code:
![изображение_2022-03-23_102152](https://user-images.githubusercontent.com/84874782/159645093-406a8ad2-489a-4b52-97f3-af04c8b6a21a.png)
error:
![image](https://user-images.githubusercontent.com/84874782/159645415-aeacd7da-232b-494c-822d-efc8a82a8908.png)

**Reply by jelni** [reply]
Your `admin` and probably all the other cogs, have non-asnc `setup` functions.

**Reply by neJok** [reply]
How can I download the version without async cogs?

**Reply by jelni** [reply]
No need to downgrade, just change the setup function definitions from `def setup(...)` to `async def setup(...)`

**Reply by neJok** [reply]
![image](https://user-images.githubusercontent.com/84874782/159685545-748d298d-d441-4678-a4d3-a526715209b2.png)
error:
![image](https://user-images.githubusercontent.com/84874782/159685482-5f2381c7-3096-414d-a69c-5a97e502beae.png)

**Reply by jelni** [reply]
Well, that's your error. You should read it: "cannot import name 'InvalidArgument' from 'discord'"

**Reply by jelni** [reply]
Read more about this change here: https://discordpy.readthedocs.io/en/master/migrating.html#removal-of-invalidargument-exception

**Reply by neJok** [reply]
Ok... How can I change these functions so that there are no errors?

**Reply by jelni** [reply]
Don't import `InvalidArgument` which doesn't exist in discord.py 2.0

**Reply by neJok** [reply]
I am not importing `InvalidArgument`

**Reply by jelni** [reply]
Oh wait, that's an error from `discord-components`. discord.py 2.0 has full support for slash commands and you should switch to using that
